esphome/components/tuya/tuya.cpp

```diff
@@ -321,24 +306,49 @@ void Tuya::handle_datapoint_(const uint8_t *buffer, size_t len) {
       listener.on_datapoint(datapoint);
 }
 
-void Tuya::send_command_(TuyaCommandType command, const uint8_t *buffer, uint16_t len) {
-  uint8_t len_hi = len >> 8;
-  uint8_t len_lo = len >> 0;
+void Tuya::send_raw_command_(TuyaCommand command) {
+  uint8_t len_hi = (uint8_t) command.payload.size() >> 8;
+  uint8_t len_lo = (uint8_t) command.payload.size() >> 0;
   uint8_t version = 0;
 
-  ESP_LOGV(TAG, "Sending Tuya: CMD=0x%02X VERSION=%u DATA=[%s] INIT_STATE=%u", command, version,  // NOLINT
-           hexencode(buffer, len).c_str(), this->init_state_);
+  this->last_command_timestamp_ = millis();
+
+  ESP_LOGV(TAG, "Sending Tuya: CMD=0x%02X VERSION=%u DATA=[%s] INIT_STATE=%u", command.cmd, version,  // NOLINT
+           hexencode(command.payload.data(), command.payload.size()).c_str(), this->init_state_);
```

Comment:
This can be simplified as the `hexencode` function has an override to accept a vector directly
```suggestion
           hexencode(command.payload).c_str(), this->init_state_);
```